Firestore/core/test/firebase/firestore/local/lru_garbage_collector_test.cc

```diff
@@ -0,0 +1,740 @@
+    Document doc3 = CacheADocumentInTransaction();
+    expected_retained.insert(doc3.key());
+    AddDocument(doc3.key(), query_data.target_id());
+  });
+
+  // cache another document and prepare a mutation on it.
+  persistence_->Run("queue a mutation", [&] {
+    Document doc4 = CacheADocumentInTransaction();
+    mutations.push_back(MutationForDocument(doc4.key()));
+    expected_retained.insert(doc4.key());
+  });
+
+  // Insert the mutations. These operations don't have a sequence number, they
+  // just serve to keep the mutated documents from being GC'd while the
+  // mutations are outstanding.
+  persistence_->Run("actually register the mutations", [&] {
+    Timestamp write_time = Timestamp::Now();
+    mutation_queue_->AddMutationBatch(write_time, {}, std::move(mutations));
+  });
+
+  // Mark 5 documents eligible for GC. This simulates documents that were
+  // mutated then ack'd. Since they were ack'd, they are no longer in a mutation
+  // queue, and there is nothing keeping them alive.
+  std::unordered_set<DocumentKey, DocumentKeyHash> to_be_removed;
+  persistence_->Run("add orphaned docs (previously mutated, then ack'd)", [&] {
+    for (int i = 0; i < 5; i++) {
+      Document doc = CacheADocumentInTransaction();
+      to_be_removed.insert(doc.key());
+      MarkDocumentEligibleForGcInTransaction(doc.key());
+    }
+  });
+
+  // We expect only the orphaned documents, those not in a mutation or a target,
+  // to be removed. use a large sequence number to remove as much as possible
+  int removed = RemoveOrphanedDocuments(1000);
+  ASSERT_EQ(to_be_removed.size(), removed);
+  persistence_->Run("verify", [&] {
+    for (const DocumentKey& key : to_be_removed) {
+      ASSERT_EQ(document_cache_->Get(key), absl::nullopt);
+      ASSERT_FALSE(query_cache_->Contains(key));
+    }
+    for (const DocumentKey& key : expected_retained) {
+      ASSERT_NE(document_cache_->Get(key), absl::nullopt)
+          << "Missing document " << key.ToString().c_str();
+    }
+  });
+  persistence_->Shutdown();
+}
+
+// TODO(gsoltis): write a test that includes limbo documents
+
+TEST_P(LruGarbageCollectorTest, RemoveTargetsThenGC) {
+  // Create 3 targets, add docs to all of them
+  // Leave oldest target alone, it is still live
+  // Remove newest target
+  // Blind write 2 documents
+  // Add one of the blind write docs to oldest target (preserves it)
+  // Remove some documents from middle target (bumps sequence number)
+  // Add some documents from newest target to oldest target (preserves them)
+  // Update a doc from middle target
+  // Remove middle target
+  // Do a blind write
+  // GC up to but not including the removal of the middle target
+  //
+  // Expect:
+  // All docs in oldest target are still around
+  // One blind write is gone, the first one not added to oldest target
+  // Documents removed from middle target are gone, except ones added to oldest
+  // target Documents from newest target are gone, except
```

Comment:
Either there's extra words in this sentence, or it's unfinished. It seems like `are gone, except` is a copy-paste error and should be removed.

Reply:
I think this was incomplete. I've reconstructed it based on what the test is doing.

I've also added bullets and full stops to make it clearer that these are each separate items.